src/instrument.ts

```diff
@@ -93,6 +95,30 @@ interface traceConfig {
 const SPAN_NAMESPACE_PREFIX = 'CloudSpanner'; // TODO: discuss & standardize this prefix.
 export {SPAN_NAMESPACE_PREFIX, traceConfig};
 
+const {
+  AsyncHooksContextManager,
+} = require('@opentelemetry/context-async-hooks');
+
+/*
+ * This function ensures that async/await functions correctly by
+ * checking if context.active() returns an invalid/unset context
+ * and if so, sets a global AsyncHooksContextManager.
+ */
```

Comment:
```suggestion
/**
 * Ensures that an appropriate Context Manager is set to support proper
 * context propagation when using async/await, particularly in OpenTelemetry.
 * 
 * OpenTelemetry's context system relies on `context.active()` to retrieve the 
 * current active context. However, if this returns the default `ROOT_CONTEXT`, 
 * it indicates that no context has been previously set. Without a valid context, 
 * async/await calls will fail to propagate context correctly, which can cause 
 * issues in distributed tracing and telemetry data collection.
 * 
 * This function checks if the active context is invalid (i.e., `ROOT_CONTEXT`). 
 * If so, it resets the Context Manager by disabling any prior manager and then 
 * enabling a new instance of `AsyncHooksContextManager` to properly track async
 * operations and context propagation.
 * 
 * The goal is to ensure that users' code works seamlessly with OpenTelemetry, 
 * without requiring them to modify their existing code to manually manage context.
 * 
 * Further details:
 * - OpenTelemetry context documentation: https://opentelemetry.io/docs/languages/js/context/#active-context
 * - Related issue: https://github.com/googleapis/nodejs-spanner/issues/2146
 */
```

Reply:
@odeke-em I generated this documentation using AI, please review once more. But we need a more structured documentation

Reply:
Some of this AI generated documentation is not correct like 
```
 * The goal is to ensure that users' code works seamlessly with OpenTelemetry, 
* without requiring them to modify their existing code to manually manage context.
```	

it's nothing about manually managing context but setting a global context manager.